<ui version="4.0" >
 <class>icfpc_guiClass</class>
 <widget class="QMainWindow" name="icfpc_guiClass" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>536</width>
    <height>388</height>
   </rect>
  </property>
  <property name="windowTitle" >
   <string>icfpc_gui</string>
  </property>
  <widget class="QWidget" name="centralWidget" >
   <layout class="QVBoxLayout" name="verticalLayout" >
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout" >
      <property name="sizeConstraint" >
       <enum>QLayout::SetFixedSize</enum>
      </property>
      <property name="rightMargin" >
       <number>0</number>
      </property>
      <item>
       <widget class="QLabel" name="label" >
        <property name="maximumSize" >
         <size>
          <width>46</width>
          <height>14</height>
         </size>
        </property>
        <property name="text" >
         <string>delay, ms</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="timeoutSpinBox" >
        <property name="minimumSize" >
         <size>
          <width>81</width>
          <height>20</height>
         </size>
        </property>
        <property name="maximumSize" >
         <size>
          <width>81</width>
          <height>20</height>
         </size>
        </property>
        <property name="maximum" >
         <number>1000000</number>
        </property>
        <property name="singleStep" >
         <number>100</number>
        </property>
        <property name="value" >
         <number>0</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_2" >
        <property name="text" >
         <string>Zoom</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="zoomSlider" >
        <property name="minimum" >
         <number>-200</number>
        </property>
        <property name="maximum" >
         <number>400</number>
        </property>
        <property name="value" >
         <number>50</number>
        </property>
        <property name="orientation" >
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="SimulationWidget" native="1" name="space" >
      <property name="sizePolicy" >
       <sizepolicy vsizetype="Expanding" hsizetype="Expanding" >
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="palette" >
       <palette>
        <active>
         <colorrole role="WindowText" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>0</red>
            <green>0</green>
            <blue>0</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Button" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>255</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Light" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>255</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Midlight" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>255</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Dark" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>127</red>
            <green>127</green>
            <blue>127</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Mid" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>170</red>
            <green>170</green>
            <blue>170</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Text" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>0</red>
            <green>0</green>
            <blue>0</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="BrightText" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>255</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="ButtonText" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>0</red>
            <green>0</green>
            <blue>0</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Base" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>255</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Window" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>255</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Shadow" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>0</red>
            <green>0</green>
            <blue>0</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="AlternateBase" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>255</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="ToolTipBase" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>255</red>
            <green>255</green>
            <blue>220</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="ToolTipText" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>0</red>
            <green>0</green>
            <blue>0</blue>
           </color>
          </brush>
         </colorrole>
        </active>
        <inactive>
         <colorrole role="WindowText" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>0</red>
            <green>0</green>
            <blue>0</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Button" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>255</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Light" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>255</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Midlight" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>255</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Dark" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>127</red>
            <green>127</green>
            <blue>127</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Mid" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>170</red>
            <green>170</green>
            <blue>170</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Text" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>0</red>
            <green>0</green>
            <blue>0</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="BrightText" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>255</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="ButtonText" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>0</red>
            <green>0</green>
            <blue>0</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Base" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>255</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Window" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>255</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Shadow" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>0</red>
            <green>0</green>
            <blue>0</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="AlternateBase" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>255</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="ToolTipBase" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>255</red>
            <green>255</green>
            <blue>220</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="ToolTipText" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>0</red>
            <green>0</green>
            <blue>0</blue>
           </color>
          </brush>
         </colorrole>
        </inactive>
        <disabled>
         <colorrole role="WindowText" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>127</red>
            <green>127</green>
            <blue>127</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Button" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>255</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Light" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>255</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Midlight" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>255</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Dark" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>127</red>
            <green>127</green>
            <blue>127</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Mid" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>170</red>
            <green>170</green>
            <blue>170</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Text" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>127</red>
            <green>127</green>
            <blue>127</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="BrightText" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>255</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="ButtonText" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>127</red>
            <green>127</green>
            <blue>127</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Base" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>255</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Window" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>255</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Shadow" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>0</red>
            <green>0</green>
            <blue>0</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="AlternateBase" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>255</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="ToolTipBase" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>255</red>
            <green>255</green>
            <blue>220</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="ToolTipText" >
          <brush brushstyle="SolidPattern" >
           <color alpha="255" >
            <red>0</red>
            <green>0</green>
            <blue>0</blue>
           </color>
          </brush>
         </colorrole>
        </disabled>
       </palette>
      </property>
      <property name="autoFillBackground" >
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar" >
   <property name="geometry" >
    <rect>
     <x>0</x>
     <y>0</y>
     <width>536</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File" >
    <property name="title" >
     <string>&amp;File</string>
    </property>
    <addaction name="action_Open" />
    <addaction name="action_Save_Trace" />
    <addaction name="actionE_xit" />
   </widget>
   <widget class="QMenu" name="menu_Play" >
    <property name="title" >
     <string>&amp;Play</string>
    </property>
    <addaction name="action_Start" />
    <addaction name="actionS_top" />
   </widget>
   <addaction name="menu_File" />
   <addaction name="menu_Play" />
  </widget>
  <widget class="QToolBar" name="mainToolBar" >
   <attribute name="toolBarArea" >
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak" >
    <bool>false</bool>
   </attribute>
   <addaction name="action_Open" />
   <addaction name="action_Start" />
   <addaction name="actionPause" />
   <addaction name="actionS_top" />
  </widget>
  <widget class="QStatusBar" name="statusBar" />
  <action name="action_Open" >
   <property name="text" >
    <string>&amp;Open</string>
   </property>
  </action>
  <action name="action_Save_Trace" >
   <property name="text" >
    <string>&amp;Save Trace</string>
   </property>
  </action>
  <action name="actionE_xit" >
   <property name="text" >
    <string>E&amp;xit</string>
   </property>
  </action>
  <action name="action_Start" >
   <property name="text" >
    <string>&amp;Start</string>
   </property>
  </action>
  <action name="actionS_top" >
   <property name="text" >
    <string>S&amp;top</string>
   </property>
  </action>
  <action name="actionPause" >
   <property name="text" >
    <string>Pause</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11" />
 <customwidgets>
  <customwidget>
   <class>SimulationWidget</class>
   <extends>QWidget</extends>
   <header>simulationwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="icfpc_gui.qrc" />
 </resources>
 <connections/>
</ui>
